### PR TITLE
[SPARK-43943][SQL][PYTHON][CONNECT] Add SQL math functions to Scala and Python

### PR DESCRIPTION
**Pull request by zhengruifeng**
### What changes were proposed in this pull request?
Add following functions:

* ceiling
* e
* pi
* ln
* negative
* positive
* power
* sign
* std
* width_bucket

to:

* Scala API
* Python API
* Spark Connect Scala Client
* Spark Connect Python Client

This PR also adds `negate` (which already exists in Scala API and SCSC) to Python API and SCPC.

### Why are the changes needed?
for parity

### Does this PR introduce _any_ user-facing change?
yes, new functions

### How was this patch tested?
added ut / doctest

### CODE REVIEW

**Review by HyukjinKwon — APPROVED** [review]

**Comment by HyukjinKwon** [comment]
oh also might need to put them in Python reference doc `.rst` file

File: python/docs/source/reference/pyspark.sql/functions.rst [hunk 1]
```diff
@@ -64,31 +64,39 @@ Math Functions
...
     floor
     hex
     unhex
     hypot
+    ln
     log
     log10
     log1p
     log2
+    negative
```
Comment (zhengruifeng) [inline python/docs/source/reference/pyspark.sql/functions.rst:86]: @HyukjinKwon they were added here. except the `negate`, I will add it.

BTW, in the last commit I add `negate` (which only existed in scala) to the python side for the parity between py and scala

**Comment by HyukjinKwon** [comment]
LGTM

**Comment by zhengruifeng** [comment]
@HyukjinKwon it seems that `sql - other` start failing again ...

**Comment by zhengruifeng** [comment]
on the sql side, this PR only touch `MathFunctionsSuite` in `sql - slow`

image


the failure in `sql - other` should be unrelated, i am going to merge it now

**Comment by zhengruifeng** [comment]
merged to master

**Comment by LuciferYang** [comment]
> @HyukjinKwon it seems that `sql - other` start failing again ...

I will investigate tomorrow, a little late today

There is one case that may have been failed after this merged , and I try to fix it in https://github.com/apache/spark/pull/41519

**Pull request closed without merging**